src/main/java/com/conveyal/analysis/controllers/BundleController.java

```diff
@@ -215,38 +218,32 @@ private Bundle create (Request req, Response res) {
                         fileStorage.moveIntoStorage(gtfsCache.getFileKey(feedSummary.bundleScopedFeedId, "db"), tempDbFile);
                         fileStorage.moveIntoStorage(gtfsCache.getFileKey(feedSummary.bundleScopedFeedId, "db.p"), tempDbpFile);
                         fileStorage.moveIntoStorage(gtfsCache.getFileKey(feedSummary.bundleScopedFeedId, "zip"), feedFile);
-
-                        // Increment feeds complete for the progress handler
-                        bundle.feedsComplete += 1;
-
-                        // Done in a loop the nonce and updatedAt would be changed repeatedly
-                        Persistence.bundles.modifiyWithoutUpdatingLock(bundle);
-                        progressListener.increment();
                     }
+                    // Set legacy progress field to indicate that all feeds have been loaded.
+                    bundle.feedsComplete = bundle.totalFeeds;
 
                     // TODO Handle crossing the antimeridian
                     bundle.north = bundleBounds.getMaxY();
                     bundle.south = bundleBounds.getMinY();
                     bundle.east = bundleBounds.getMaxX();
                     bundle.west = bundleBounds.getMinX();
                 }
-
                 writeManifestToCache(bundle);
                 bundle.status = Bundle.Status.DONE;
               } catch (Throwable t) {
-                // This catches any error while processing a feed with the GTFS Api and needs to be more
-                // robust in bubbling up the specific errors to the UI. Really, we need to separate out the
-                // idea of bundles, track uploads of single feeds at a time, and allow the creation of a
-                // "bundle" at a later point. This updated error handling is a stopgap until we improve that
-                // flow.
                 LOG.error("Error creating bundle", t);
                 bundle.status = Bundle.Status.ERROR;
                 bundle.statusText = ExceptionUtils.shortAndLongString(t);
+                // Rethrow the problem so the task scheduler will attach it to the task with state ERROR.
+                // Eventually this whole catch and finally clause should be handled generically up in the task scheduler.
+                throw t;
               } finally {
+                // ID and create/update times are assigned here when we push into Mongo.
                 Persistence.bundles.modifiyWithoutUpdatingLock(bundle);
               }
         }));
-
+        // TODO do we really want to return the bundle here? It should not be needed until the background work is done.
+        // We could instead return the WorkProduct instance (or BaseModel instance or null) from TaskActions.
         return bundle;
```

Comment:
Should we return the `Task` (or `ApiTask`)? We wouldn't currently use that in the UI, but that might be a common pattern that we could use in the future. "The progress for the action you just took will be tracked via this task"

Reply:
It seems sufficient to return only an HTTP status code "Accepted". We could also return the task and potentially even a workProduct (it turns out we have to create them and assign IDs before processing a task to allow exception handling one level up). But unless we've got a specific need for those return values, I think HTTP 202 is best.